bibcheck/plugins/fermilab_coll.py

```diff
@@ -25,10 +25,19 @@ def check_records(records):
     for record in records:
         in_coll = False
         has_rn = False
-        for (tag, _, _), value in record.iterfields(['037__a', '037__z', '980__a']):
-            if tag[0:5] == '037__' and 'fermilab' in value.lower():
+        slidepost = False
+        hep_pos = None
+        for pos, value in record.iterfields(['037__a', '037__z', '980__a']):
+            if pos[0][0:5] == '037__' and 'fermilab' in value.lower():
                 has_rn = True
-            if tag == '980__a' and value == 'Fermilab':
-                in_coll = True
+                if 'slides' in value.lower() or 'poster' in value.lower():
+                    slidepost = True
+            if pos[0] == '980__a':
+                if value == 'Fermilab':
+                    in_coll = True
+                elif value == 'HEP':
+                    hep_pos = pos
         if has_rn and not in_coll:
             record.add_field('980__', '', subfields=[('a', 'Fermilab')])
+        if slidepost and hep_pos:
```

Comment:
what if 980:HEP appears twice

you have `record.add_filed` before `record.delete_field`, are you sure that the `pos` hasn't changed, since the underlying record was changed in the previous step?